File: backend/app/services/navigation_service.py
from app.schemas.schemas import NavigationGoalRequest, NavigationStatusResponse


class NavigationService:
    @staticmethod
    async def get_status() -> NavigationStatusResponse:
        return NavigationStatusResponse()

    @staticmethod
    async def send_goal(goal: NavigationGoalRequest) -> dict:
        return {"status": "GOAL_ACCEPTED", "goal": {"x": goal.x, "y": goal.y, "yaw": goal.yaw}}

    @staticmethod
    async def cancel_goal() -> dict:
        return {"status": "GOAL_CANCELLED"}

    @staticmethod
    async def pause() -> dict:
        return {"status": "NAVIGATION_PAUSED"}

    @staticmethod
    async def resume() -> dict:
        return {"status": "NAVIGATION_RESUMED"}


navigation_service = NavigationService()

File: frontend/src/services/navigation.service.ts
import api from './api';
import { NavigationPose } from '../types';

export const navigationService = {
  getStatus: async (): Promise<NavigationPose> => {
    return api.get('/navigation/status');
  },

  sendGoal: async (goal: { x: number; y: number; yaw?: number }) => {
    return api.post('/navigation/goal', goal);
  },

  cancelGoal: async () => {
    return api.post('/navigation/cancel');
  },

  pause: async () => {
    return api.post('/navigation/pause');
  },

  resume: async () => {
    return api.post('/navigation/resume');
  },
};

export default navigationService;
